doc/site/modules/core/object.markdown

```diff
@@ -29,6 +29,8 @@ Compares two objects using built-in equality. This compares [value
 types](../values.html) by value, and all other objects are compared by
 identity&mdash;two objects are equal only if they are the exact same object.
 
+Note however that these operators are overridden in the `Num` class to ensure `0` and `-0` are treated as being equal in value even though they have different bit representations. Object.same(0, -0) still returns `false` because they are different objects.
```

Comment:
I would say "different values" (last two words). I wouldn't want to imply numbers are objects, since they're not (they're a primitive value).

Reply:
although Object.same on two distinct values is a weird thing... would we even expect it to return true when you're comparing two distinct _values_?

Reply:
The reason I used the word `object` is because we're in the Object class docs and, technically, everything in Wren is an object even numeric literals. On the other hand Num is a value type and so they're values as well. If you're more comfortable with `value` we can change it.

You'd expect Object.same to return false and so the last sentence is really superfluous. The only reason I put it in was to make the distinction between that method and the equals operator crystal clear but we can remove it if you like. 

Incidentally, when I was playing around with this the other day, I noticed another weird thing about negative zero. Although (-0).toString returns `-0` as you'd expect, (-0).sign returns `0` rather than `-1`. I'm not suggesting that this is wrong and we should change it but do you think we should mention it in the docs for Num.sign?

Reply:
from the `same` docs right above, yea:

> For value types, this returns true if the objects have equivalent state. In other words, numbers, strings, booleans, and ranges compare by value.

I don't think using 'object' to refer to values is clearer, it adds questions. I don't think same needs a mention.

I also feel a bit weird about Num details being in the `==` operator documentation, but regardless I'd simplify this either way. It could be relevant to mention it in the Num docs, where it documents `==` specifically to call this out. That's where I'd look to answer questions about Num.

> Note: `Num` overrides these operators so that `0 == -0` returns `true`.